plugin.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const PLUGIN_ID_KEY = '___shelljs_plugin_id___';
```

Comment:
Let me explain about the whole plugin id thing:

There are two versions of every command: The one that's actually exposed as a property of `$`, and the internal representation, which has some other metadata and state.

This is a cheap and easy way to link them. The thing that's actually exposed gets a non-enumerable, non-writable, non-configurable property called `___shelljs_plugin_id___`, which is set to a number. Then, we have an array of internal plugin representations [here](#diff-18e559ebd3cc204a52bc1dc3db2adf8bR17), where `plugins[i].id === i`.